2001-11-24  Sam Hartman  <hartmans@mit.edu>

	* kpropd.c (do_standalone): Default to KPROP port
	* kprop.c (open_connection): Default to 754 rather than an error,
	Debian bug #120010 

Kerberos RT #3268
Debian bug #120010

--- krb5-1.4.2.orig/src/slave/kprop.c
+++ krb5-1.4.2/src/slave/kprop.c
@@ -346,13 +346,11 @@
 	if(!port) {
 		sp = getservbyname(KPROP_SERVICE, "tcp");
 		if (sp == 0) {
-			(void) strncpy(Errmsg, KPROP_SERVICE, ErrmsgSz - 1);
-			Errmsg[ErrmsgSz - 1] = '\0';
-			(void) strncat(Errmsg, "/tcp: unknown service", ErrmsgSz - 1 - strlen(Errmsg));
-			*fd = -1;
-			return(0);
+		    my_sin.sin_port = htons(KPROP_PORT);
+		
+		} else {
+		    my_sin.sin_port = sp->s_port;
 		}
-		my_sin.sin_port = sp->s_port;
 	} else
 		my_sin.sin_port = port;
 	s = socket(AF_INET, SOCK_STREAM, 0);
--- krb5-1.4.2.orig/src/slave/kprop.h
+++ krb5-1.4.2/src/slave/kprop.h
@@ -29,6 +29,7 @@
 #define KPROP_SERVICE_NAME "host"
 #define TGT_SERVICE_NAME "krbtgt"
 #define KPROP_SERVICE "krb5_prop"
+#define KPROP_PORT 754
 
 #define KPROP_PROT_VERSION "kprop5_01"
 
--- krb5-1.4.2.orig/src/slave/kpropd.c
+++ krb5-1.4.2/src/slave/kpropd.c
@@ -193,9 +193,9 @@
 		sp = getservbyname(KPROP_SERVICE, "tcp");
 		if (sp == NULL) {
 			com_err(progname, 0, "%s/tcp: unknown service", KPROP_SERVICE);
-			exit(1);
+			my_sin.sin_port = htons(KPROP_PORT);
 		}
-		my_sin.sin_port = sp->s_port;
+		else my_sin.sin_port = sp->s_port;
 	} else {
 		my_sin.sin_port = port;
 	}
